cpp/src/arrow/type_test.cc

```diff
@@ -398,6 +563,71 @@ TEST(TestFieldRef, Basics) {
   EXPECT_THAT(FieldRef("beta").FindAll(s), ElementsAre(FieldPath{1}, FieldPath{3}));
 }
 
+TEST(TestFieldRef, FindAllForTable) {
```

Comment:
I don't think the extra `FieldRef` tests need to be included here (again, maybe in a separate PR).

In any case, you wouldn't  need to go through all this trouble to create the `Table`/`RecordBatch` since `FieldRef::FindAll` doesn't actually care about the values (eventually, all the overloads just redirect to the overload for `FieldVector`). The `Find*` methods simply return `FieldPath`s which can _then_ be used to extract any data.

Reply:
In my opinion, it seems ok that we have test cases for `FindAll` function for specific types. 
The question is about the coverage. So, if @zeroshade @westonpace agree with these test cases removal I'll do it

Reply:
I agree it would be worthwhile to test all the overloads in general. We could probably do it pretty efficiently with a test fixture.

Reply:
Let's create a github issue for the generalization of these tests (I agree that is a good idea and easier to maintain).  If we want to leave these tests in for now then we should mention (in the issue) that these tests can be replaced when the generic approach is implemented.